### PR TITLE
Improve http service

### PR DESCRIPTION
**Pull request by sixtusagbo**
## PR Description

I was trying to mock the request made on the app by mocking the `http.request()` method because parts of it that makes the request using `http.Client` object and it needs to be replaced to avoid making actual requests in the case of unit and widget tests.

I went through the code in `http_service.dart` and it does not accept as a parameter `http.Client` object. And `collectionStateNotifierProvider` does not accept `http.Client` object as a parameter too.
If it accepts a client object (as a parameter), anyone will be able to override the client when testing in order mock the http request or to unit test the service `http_service.dart`.

## Related Issues

- Related Issue #118 

### Checklist
- [x] I have gone through the [contributing guide](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md)
- [x] I have updated my branch and synced it with project `main` branch before making this PR
- [x] I have run the tests (`flutter test`) and all tests are passing

## Added/updated tests?
_We encourage you to add relevant test cases._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests have not been included_

### CODE REVIEW

**Comment by ashitaprasad** [comment]
We will currently not make any changes to the http_service. Instead of mocking the request you can send an actual request in the tests as these apis are also maintained by us.

**Comment by sixtusagbo** [comment]
Okay, I've removed the mocks. I am sending actual requests now. I've completed test cases for `codePaneVisibleStateProvider`

File: test/providers/ui_providers_test.dart [hunk 1]
```diff
@@ -336,4 +343,274 @@ void main() {
...
+
+      // Verify that NotSentWidget is visible
+      expect(find.byType(NotSentWidget), findsOneWidget);
+
+      // Add some data in URLTextField
+      Finder field = find.descendant(
+        of: find.byType(URLField),
+        matching: find.byType(TextFormField),
+      );
+      const url = 'https://api.foss42.com/country/codes';
```
Comment (animator) [inline test/providers/ui_providers_test.dart:544]: Put URL const in https://github.com/foss42/apidash/blob/main/test/test_consts.dart

**Comment by animator** [comment]
Just like we have defined example request models in https://github.com/foss42/apidash/blob/main/test/request_models.dart, you need to add constants like the URL used in tests in https://github.com/foss42/apidash/blob/main/test/test_consts.dart
Also, use `api.apidash.dev` not `api.foss42.com`

File: test/providers/ui_providers_test.dart (hunk 1 above)
Comment (sixtusagbo) [reply test/providers/ui_providers_test.dart]: I've added it

Comment (sixtusagbo) [reply test/providers/ui_providers_test.dart]: I've added it

**Comment by sixtusagbo** [comment]
Okay Sir, I just updated the files now.

**Comment by animator** [comment]
LGTM 🚀

**Pull request merged**